internal/bpf/bpfWrapper.c

```diff
@@ -150,60 +154,60 @@ int Clean_bpf(char *ifname) {
 	Log_Info("%s: starting removal of xdp program on interface %s (%d)", __FUNCTION__, ifname,
 		 if_index);
 
-	err = bpf_set_link_xdp_fd(if_index, fd, XDP_FLAGS_UPDATE_IF_NOEXIST);
+	mp = xdp_multiprog__get_from_ifindex(if_index);
+	if (!mp) {
+		Log_Error("%s: unable to receive correct multi_prog reference : %s", __FUNCTION__,
+			  mp);
+		return -1;
+	}
+
+	err = xdp_multiprog__detach(mp);
 	if (err) {
-		if (err == EBUSY_CODE_WARNING) {
-			// unloading of XDP program found to return EBUSY error of -16 on certain
-			// host libbpf versions. doesn't break functionality and this problem is
-			// being investigated.
-			Log_Warning("%s: Removal of xdp program is reporting error code: (%d)",
-				    __FUNCTION__, err);
-		} else {
-			Log_Error("%s: Removal of xdp program failed, returned: (%d)", __FUNCTION__,
-				  err);
-			return 1;
-		}
+		Log_Error("%s: Removal of xdp program failed, returned: "
+			  "returned: %d",
+			  __FUNCTION__, err);
+		return -1;
 	}
 
 	Log_Info("%s: removed xdp program from interface %s (%d)", __FUNCTION__, ifname, if_index);
 	return 0;
 }
 
 int Load_attach_bpf_xdp_pass(char *ifname) {
-	int prog_fd = -1, err, ifindex;
-	char *filename = "/afxdp/xdp_pass.o";
-	struct bpf_object *obj;
-	__u32 xdp_flags = XDP_FLAGS_UPDATE_IF_NOEXIST | XDP_FLAGS_DRV_MODE;
-
-	Log_Info("%s: disovering if_index for interface %s", __FUNCTION__, ifname);
-
-	ifindex = if_nametoindex(ifname);
-	if (!ifindex) {
-		Log_Error("%s: if_index not valid: %s", __FUNCTION__, ifname);
-		return -1;
-	} else {
-		Log_Info("%s: if_index for interface %s is %d", __FUNCTION__, ifname, ifindex);
-	}
-
-	Log_Info("%s: starting setup of xdp-pass program on "
-		 "interface %s (%d)",
-		 __FUNCTION__, ifname, ifindex);
-
-	/* Load the BPF program */
-	err = bpf_prog_load(filename, BPF_PROG_TYPE_XDP, &obj, &prog_fd);
-	if (err < 0) {
-		Log_Error("%s: Couldn't load BPF-OBJ file(%s)\n", __FUNCTION__, filename);
-		return -1;
-	}
-
-	/* Attach the program to the interface at the xdp hook */
-	err = bpf_set_link_xdp_fd(ifindex, prog_fd, xdp_flags);
-	if (err < 0) {
-		Log_Error("%s: Couldn't attach the XDP PASS PROGRAM TO %s\n", __FUNCTION__, ifname);
-		return -1;
-	}
-
-	Log_Info("%s: xdp-pass program loaded on %s (%d)", __FUNCTION__, ifname, ifindex);
+	// int prog_fd = -1, err, ifindex;

```

Comment:
An example of how to program this can be found here: 
https://github.com/xdp-project/xdp-tutorial/blob/774189980a8a67f8d4947a595b9afe6059935274/advanced03-AF_XDP/af_xdp_user.c#L542

roughly:

```c
               DECLARE_LIBBPF_OPTS(bpf_object_open_opts, opts);
    	       DECLARE_LIBXDP_OPTS(xdp_program_opts, xdp_opts, 0);

    	       prog = xdp_program__open_file(cfg.filename,
						  NULL, &opts);
		err = libxdp_get_error(prog);
		if (err) {
			libxdp_strerror(err, errmsg, sizeof(errmsg));
			fprintf(stderr, "ERR: loading program: %s\n", errmsg);
			return err;
		}

		err = xdp_program__attach(prog, cfg.ifindex, cfg.attach_mode, 0);
		if (err) {
			libxdp_strerror(err, errmsg, sizeof(errmsg));
			fprintf(stderr, "Couldn't attach XDP program on iface '%s' : %s (%d)\n",
				cfg.ifname, errmsg, err);
			return err;
		}
```